app/services/course/assessment/question/programming/cpp/cpp_package_service.rb

```diff
@@ -0,0 +1,248 @@
+# frozen_string_literal: true
+class Course::Assessment::Question::Programming::Cpp::CppPackageService < \
```

Comment:
yeah am currently looking into how to get the c/c++ autograder to work for multiple file submission. Should that be included in this PR as well?
As for the refactoring, it seems likely that there will be quite a bit of repeated code across the 3 packages so will be done eventually.